syntax = "proto3";

message Route {
    bytes route_id = 1;
    int64 timestamp = 2;
    string saddr = 3;
    string daddr = 4;
}



message Message {
    message Metadata {
        repeated Route route = 1;
        bytes src = 2;
        bytes dst = 3;
        optional bytes gossip = 4;
        optional bytes syn = 5;
        optional bytes ack = 6;
        optional string topic = 7;
    }

    enum Type {
        GOSSIP = 0;
        SYN = 1;
        ACK = 2;
        PUB = 3;
        SUB = 4;
        REQ = 5;
        RES = 6;
    }

    bytes message_id = 1;
    Type message_type = 2;
    Metadata metadata = 3;
    bytes payload = 4;
}
